### ISSUE TITLE
recover original read names from SRA file

### ISSUE THREAD

**Issue by map2085** [ncbi/sra-tools#141]
The read names generated by Illumina machines are structured and provide useful information about the generation of the read on the flowcell.  This information is critical for detecting optical duplicates, e.g. with the [MarkDuplicates](https://broadinstitute.github.io/picard/command-line-overview.html#MarkDuplicates) tool from the popular PICARD software suite, which parses the read names generated by the Illumina machine into coordinates on the flow cell. These coordinates are analyzed to determine if two reads are likely to be optical duplicates or not.

The reads printed by the `fastq-dump` tool are generic and do not contain such information, making analysis of Optical Duplicates impossible.

Thus, the SRA convention for storing read names, that is, replacing them by an integer called the "spot id", loses crucial information.

Is there any way to recover this important data?

Thank you
_______________________
I recognize that [another issue](https://github.com/ncbi/sra-tools/issues/115) raises a similar question, but that user was concerned about customized text from the comments field of the FASTQ file related to barcoding unique to the experiment in question, which is understandably not officially formatted and therefore not necessarily supportable. That user was also asking about data which is not widespread & standardized in the same way that the Illumina read name convention is widespread & standardized. And I am here asking about the actual read name in the FASTQ line, not the comment field.

**Comment by kwrodarmer** [comment]
You are a victim of loose activity with read names. The SRA recorded read names and extracted critical information from them for years. The ETL error rate during that time was quite high and the cost to both submitters and the US taxpayer for all of the manual intervention, analysis and special work-around coding and debugging for us to handle _an opaque read id_ that was overloaded and abused to include information clearly critical to someone, but neither standard nor predictable - made it so that we eventually dropped the effort. Had people stuck with manufacturer supplied names and had the manufacturers published consistent, well-formed and standardized naming conventions, we would have been able to sustain it.

As a result, we do not currently provide or have available to us a means of providing access to the original names. If manufacturers are generating _any_ critical information, it would seem like a good candidate to place in an actual field dedicated to describing such information, rather than magically encoded within an opaque string.

That said, we understand and share your frustration and are looking at ways to enable you to obtain this (and other types of) information in the future.

**Comment by map2085** [comment]
The [illumina read name format ](http://support.illumina.com/content/dam/illumina-support/help/BaseSpaceHelp_v2/Content/Vault/Informatics/Sequencing_Analysis/BS/swSEQ_mBS_FASTQFiles.htm) is standard and consistent for years. It seems to be so informative and consistent over the years (only a difference in Casava pre/post version 1.8) that a [major tool](https://broadinstitute.github.io/picard/command-line-overview.html#MarkDuplicates) was written to interpret it, which is widely used in major genomics pipelines.

**Comment by kwrodarmer** [comment]
What is it that you are suggesting? It sounds as if you are suggesting that Illumina has a stable and parseable name format. We ingest **several dozen terabytes** of submitted data on any given day, processed with [major tools](https://github.com/ncbi/sra-tools) that possess sophisticated parsers capable of recognizing well-defined names. How would you like us to handle submissions where the names are **not** well formed? Is this a submitter error? Are we supposed to perform fuzzy parsing? Natural language? What is your proposal?

**Comment by seandavi** [comment]
Just out of curiosity, @kwrodarmer, could the submitter names be "kept" rather than "parsed"? Are there situations where the read names are so degenerate that they do not fit the FASTQ format that would break any ingestion pipeline? Not a proposal, just curious.

**Comment by map2085** [comment]
My comment was intended to illustrate the substantial importance and widespread use of the Illumina generated read names, since the earlier comment was overly dismissive of their utility and suggested that I had been gullable about their utility.

I was thinking along the lines of the suggestion by @seandavi . 

Why the focus on parsing the read name?  Does the focus on parsing the read names derive from the goal of the SRA to have maximum compression?  Are the read names not stored "as is" anywhere?

**Comment by kwrodarmer** [comment]
@map2085 - I have reread my comments looking for where I suggested you'd been gullible or that read names had no utility, but could not find it! I **do** see where I said that read names are being used to store valuable information when there is no requirement for them to conform to any particular specification.

We have processed millions of submissions from every corner of the globe using every known technology and quite frequently pre-processed by some intermediate pipeline that wants to annotate the data in some way. Where do they store their annotations? If it's FASTQ, then they use the read names, of course! Where else could they put them?

More importantly, how could we possibly be expected to deal with them? If we had had the luxury of dealing with a single manufacturer with a stable name format, things would be different. But we didn't, and this is why I said you'd been the victim of many abuses to read names that led us to stop storing them in general. It's not that we think they don't carry useful information, but that it's impossible to harmonize.

@seandavi  and @map2085 - regarding the idea of just storing the names as opaque strings rather than parsing them (or the slightly optimized case of storing unrecognized reads as-is), the read names actually take a fair amount of storage, so it comes down to an economic question. Even without read names and with the amount of compression that we get from using VDB, NCBI still has nearly 10 petabytes of expensive, hot storage dedicated to the SRA. Even if reads were only 10% of the overall run data, should we dedicate another petabyte of storage to them?

Read names are parsed both to extract important information and to allow them to be compressed. When they cannot be parsed, they are quite bulky, and of course yield no information that can be harmonized. We do not just try to recreate an originally submitted file, but emit files in several formats from a single source. This would be impossible without some level of successful read name parsing.
